### PR TITLE
Implement customizable per-session persistance

### LINKED ISSUE #71: Change cookie persistence according to user choice
I'd like to give the users of my application the option to persist their session (the infamous "Remember me" checkbox) but I'm not sure how I should proceed. 

Can I just change the Cookie.Persist when the user logs in? Should I save that option in their context to pass it again when I fire up RenewToken()? If I change the persist option, is it possible that other users will be affected?

Btw thank you for scs!

### PR DESCRIPTION
**Pull request by lafriks**
Fixes #71

### CODE REVIEW

**Comment by lafriks** [comment]
Conflict resolved. Is there anything else needs to be done?

File: codec.go [hunk 1]
```diff
@@ -9,18 +9,20 @@ import (
 // Codec is the interface for encoding/decoding session data to and from a byte
 // slice for use by the session store.
 type Codec interface {
-	Encode(deadline time.Time, values map[string]interface{}) ([]byte, error)
-	Decode([]byte) (deadline time.Time, values map[string]interface{}, err error)
+	Encode(deadline time.Time, persit bool, values map[string]interface{}) ([]byte, error)
```
Comment (citruspi) [inline codec.go]: Hey @lafriks, just a heads up, there's 4 typos in this file on lines 12, 13, 18, and 25.

In each instance `persist` is misspelled as `persit`.

Comment (lafriks) [reply codec.go]: Fixed

**Comment by lafriks** [comment]
@alexedwards friendly ping ;)

**Comment by 1gm** [comment]
This would be a breaking change to the Codec interface and require a major version bump. Maybe this and #81 belong in v3?

**Comment by alexedwards** [comment]
Sorry, I've not been ignoring this, but considering it.

As this is a breaking change to the Store interface I wouldn't want to merge this right now anyway, and would prefer to wait until there are a number of breaking changes that can all be implemented together in one new release.

But I don't think it's the right solution to the problem anyway. It's probably better to add a single mechanism for overriding *any* setting on a per-session basis, rather than doing just the Persist setting.

I've opened issue #85 to track that.

**Pull request closed without merging**